pkg/plugins/runtime/k8s/controllers/gatewayapi/converter.go

```diff
@@ -0,0 +1,175 @@
+package controllers
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+)
+
+func serviceTagForGateway(name, namespace string) string {
+	return fmt.Sprintf("%s_%s_gateway", name, namespace)
+}
+
+func k8sToKumaHeader(header gatewayapi.HTTPHeader) *mesh_proto.GatewayRoute_HttpRoute_Filter_RequestHeader_Header {
+	return &mesh_proto.GatewayRoute_HttpRoute_Filter_RequestHeader_Header{
+		Name:  string(header.Name),
+		Value: header.Value,
+	}
+}
+
+func gapiToKumaRef(objectNamespace string, ref gatewayapi.BackendObjectReference) (map[string]string, error) {
+	// References to Services are required by GAPI to include a port
+	if ref.Port == nil {
+		return nil, errors.New("backend reference must include port")
+	}
```

Comment:
I filed https://github.com/kubernetes-sigs/gateway-api/issues/941

Reply:
will add as TODO to remove